src/main/java/net/fabricmc/loom/util/DownloadUtil.java

```diff
@@ -60,7 +61,14 @@ public static void downloadIfChanged(URL from, File to, Logger logger) throws IO
 	 * @throws IOException If an exception occurs during the process
 	 */
 	public static void downloadIfChanged(URL from, File to, Logger logger, boolean quiet) throws IOException {
-		HttpURLConnection connection = (HttpURLConnection) from.openConnection();
+		URLConnection urlConnection = from.openConnection();
+
+		if (!(urlConnection instanceof HttpURLConnection)) {
+			Files.asByteSink(to).writeFrom(urlConnection.getInputStream());
```

Comment:
No this is awful, this is for downloading files. Again please keep the PR minimal.

Reply:
I was testing custom manifests. Will remove